draft-kohbrok-mls-associated-parties.md

```diff
@@ -290,34 +229,56 @@ export the `ap_commit_base_secret` from the key schedule of the new epoch.
 Both associated party and all other group members finally derive the
 `ap_commit_secret_id` and the `ap_commit_secret`. They compare the
 `ap_commit_secret_id` to the one they received along with the ciphertext to
-ensure that everyone got the same base secret. The `ap_commit_secret` is later
-used to compute the new epoch of the associated party key schedule.
+ensure that everyone got the same base secret. The `ap_commit_secret` is used to
+compute the new epoch of the associated party key schedule as described in
+{{associated-party-key-schedule}}.
 
-TODO: We probably want a distinct WireFormat for the message with `kem_output`,
-`ciphertext and `ap_commit_secret_id`. That message should also be authenticated
-using a signature.
+Open Question: Do we want a completely new message format here? It would
+probably be easier to extend the `ContentType` enum, which would allow this
+document to re-use the PublicMessage semantics. The content should then consist
+of `kem_output`, `ciphertext and `ap_commit_secret_id`.
 
-## Randomness contributions from associated parties
+# Associated party key schedule
 
-Associated parties can also contribute secret key material to the shared key
-schedule. An associated party can do so by sampling fresh randomness and
-HPKE-encrypting it to the public key in the group’s ExternalPub GroupInfo
-extension (if present). The associated party then includes the resulting HPKE
-ciphertext in an AssociatedPartySecret proposal and sends it to the group. That
-secret is injected into the associated party key schedule in the epoch in which
-it is committed.
+The secrets exported from the MLS key schedule and sent to the APs as described
+in {{exporting-the-ap-secrets}} drive the AP key schedule, a simple key schedule
+that mimics the MLS key schedule. In each epoch, the newly added key material is
+combined with the key material from the previous epoch to derive the `ap_init`
+for the next epoch and the `ap_exporter`. The `ap_exporter` keys a puncturable
```

Comment:
Verb missing here:
"The `ap_exporter` keys [???] a puncturable"

Reply:
Heh. "keys" was meant as a verb, but it's probably a bit unusual. I'll change it to something that makes the sentence easier to parse.